src/models/class.model.js

```diff
@@ -0,0 +1,41 @@
+const mongoose = require('mongoose');
+const Subject = require('./enumerations/subject.model.js');
+const { toJSON, paginate } = require('./plugins');
+
+const classSchema = mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+      trim: true
+    },
+    language: {
+      type: String,
+      required: true,
+      trim: true
+    },
+    subject: {
+      type: Subject,
+      required: true
+    },
+    teacher: {
+      type: mongoose.SchemaTypes.ObjectId,
+      ref: 'User'
+    },
+    students: [{
+      student: {
+        type: mongoose.SchemaTypes.ObjectId,
+        ref: 'User'
+      }
+    }]
```

Comment:
you can change this to
```js
students: [{
  type: mongoose.SchemaTypes.ObjectId,
  ref: 'User'
}]
```
since it should just be an array of ids and not an array of objects containing ids